### PR TITLE
Adding GVRPicker.pickVisible

### PR DESCRIPTION
**Pull request by NolaDonato**
The picker internally has access to the list of visible objects compiled
by the renderer each frame. This function provides access to that list from Java.

This feature is added in response to issue #762 

GearVRf-DCO-1.0-Signed-off-by: Nola Donato nola.donat@samsung.com

### CODE REVIEW

**Comment by rahul27** [comment]
I assumed that we were planning to move away from the polling model proposed in this change to an event based model for picking. 

If we were to implement this change to fit the events model then it would make sense to create a new type of picker like a box or frustum picker to the camera rig to listen for IPickEvents.

**Comment by NolaDonato** [comment]
OK I added GVRFrustumPicker which will generate the same pick events as GVRPicker

**Comment by rahul27** [comment]
:+1: Looks good. Tried this with the eye picking sample, changed the GVRPicker to GVRFrustumPicker to test.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRFrustumPicker.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+                {
+                    GVRSceneObject sceneObj = hit.hitObject;
+                    GVRSceneObject.BoundingVolume bv = sceneObj.getBoundingVolume();
+                    center.set(bv.center, 1);
+                    center.mul(view_matrix);
+                    dir.z = bv.radius;
+                    dir.mul(view_matrix);
+                    if (!mCuller.isSphereInsideFrustum(center.x, center.y, center.z, dir.length()))
+                    {
+                        Log.d("Picker", "Picker: outside %s (%f, %f, %f) %f", sceneObj.getName(), center.x, center.y, center.z, dir.length());
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRFrustumPicker.java]: We should remove the logging. Can be done later of course.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRFrustumPicker.java [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        }
+        else
+        {
+            mCuller = new FrustumCuller(projMatrix);
+        }
+    }
+
+    public void onDrawFrame(float frameTime)
+    {
+        if (isEnabled())
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRFrustumPicker.java]: I think we should unregister the drawFrameListener if the component is disabled instead. Not something that needs immediate attention.

**Comment by liaxim** [comment]
Should we document that probably people want to add the component to ``mainScene.getMainCameraRig().getHeadTransform().getOwnerObject()``? Considering how obfuscated that looks, should we "api"-fy the common use-case?

**Comment by liaxim** [comment]
If I modify the eye-picking sample accordingly and use ``setFrustum(90, 1, znear, zfar)`` what should I be seeing? znear and zfar are retrieved from the main camera rig. I kind of expect all boards and bunnies to be red but that is not what happens.

**Comment by liaxim** [comment]
Looks good. Will merge Thursday afternoon if there are no further comments.

**Comment by NolaDonato** [comment]
setFrustum(90, 1, near, far) should show all the bunnies as red. If that is not happening it is a bug. I will look at it.

**Comment by NolaDonato** [comment]
GVRCameraRig.getHeadTransformObject is protected. I will make it public. That is the accessor for the head transform owner object.

**Pull request merged**